google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ChangeStreamMutation.java

```diff
@@ -0,0 +1,304 @@
+  public String getSourceClusterId() {
+    return this.sourceClusterId;
+  }
+
+  /** Get the commit timestamp of the current mutation. */
+  @Nonnull
+  public Timestamp getCommitTimestamp() {
+    return this.commitTimestamp;
+  }
+
+  /**
+   * Get the tie breaker of the current mutation. This is used to resolve conflicts when multiple
+   * mutations are applied to different clusters at the same time.
+   */
+  public int getTieBreaker() {
+    return this.tieBreaker;
+  }
+
+  /** Get the token of the current mutation, which can be used to resume the changestream. */
+  public String getToken() {
+    return this.token;
+  }
+
+  /** Get the low watermark of the current mutation. */
+  public Timestamp getLowWatermark() {
+    return this.lowWatermark;
+  }
+
+  /** Get the list of mods of the current mutation. */
+  @Nonnull
+  public List<Entry> getEntries() {
+    return this.entries.build();
+  }
+
+  /** Helper class to create a ChangeStreamMutation. */
+  public static class Builder {
+    private final ByteString rowKey;
+
+    private final Type type;
+
+    private final String sourceClusterId;
+
+    private final Timestamp commitTimestamp;
+
+    private final int tieBreaker;
+
+    private transient ImmutableList.Builder<Entry> entries = ImmutableList.builder();
+
+    private String token;
+
+    private Timestamp lowWatermark;
+
+    private Builder(
+        ByteString rowKey,
+        Type type,
+        String sourceClusterId,
+        Timestamp commitTimestamp,
+        int tieBreaker) {
+      this.rowKey = rowKey;
+      this.type = type;
+      this.sourceClusterId = sourceClusterId;
+      this.commitTimestamp = commitTimestamp;
+      this.tieBreaker = tieBreaker;
+    }
+
+    Builder setCell(
+        @Nonnull String familyName,
+        @Nonnull ByteString qualifier,
+        long timestamp,
+        @Nonnull ByteString value) {
+      this.entries.add(SetCell.create(familyName, qualifier, timestamp, value));
+      return this;
+    }
+
+    Builder deleteCells(
+        @Nonnull String familyName,
+        @Nonnull ByteString qualifier,
+        @Nonnull TimestampRange timestampRange) {
+      this.entries.add(DeleteCells.create(familyName, qualifier, timestampRange));
+      return this;
+    }
+
+    Builder deleteFamily(@Nonnull String familyName) {
+      this.entries.add(DeleteFamily.create(familyName));
+      return this;
+    }
+
+    public Builder setToken(@Nonnull String token) {
+      this.token = token;
+      return this;
+    }
+
+    public Builder setLowWatermark(@Nonnull Timestamp lowWatermark) {
+      this.lowWatermark = lowWatermark;
+      return this;
+    }
+
+    public ChangeStreamMutation build() {
+      Preconditions.checkArgument(
+          token != null && lowWatermark != null,
+          "ChangeStreamMutation must have a continuation token and low watermark.");
+      return new ChangeStreamMutation(this);
+    }
+  }
+
```

Comment:
Added. It will only be used by the java veneer client(if ever) so I mark it as package private. 

I don't think the user will want to use the builder directly.